graph/src/data/schema.rs

```diff
@@ -606,6 +714,192 @@ impl Schema {
             })
     }
 
+    fn validate_fulltext_directives(&self) -> Vec<SchemaValidationError> {
+        self.subgraph_schema_object_type()
+            .map_or(vec![], |subgraph_schema_type| {
+                subgraph_schema_type
+                    .directives
+                    .iter()
+                    .filter(|directives| directives.name.eq("fulltext"))
+                    .fold(vec![], |mut errors, fulltext| {
+                        self.validate_fulltext_directive_name(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_language(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_algorithm(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        self.validate_fulltext_directive_includes(fulltext)
+                            .into_iter()
+                            .for_each(|err| errors.push(err));
+                        errors
+                    })
+            })
+    }
+
+    fn validate_fulltext_directive_name(&self, fulltext: &Directive) -> Vec<SchemaValidationError> {
+        let name_value = fulltext
+            .arguments
+            .iter()
+            .find(|(key, _)| key.eq("name"))
+            .map(|(_, value)| value);
+        let name = match name_value {
+            Some(Value::String(name)) => name,
+            _ => return vec![SchemaValidationError::FulltextNameUndefined],
+        };
+
+        // Validate that fulltext.name does not overlap with any of the types
+        // TODO: Validate that each fulltext directive has a distinct name
```

Comment:
Mmh, yeah, this is a rather important check that we shouldn't leave out. It could be better placed in the `validate_fulltext_directives` method though.